python/tank/deploy/tank_commands/desktop_migration.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import textwrap
+
+from .action_base import Action
+from . import console_utils
+from ...platform import constants
+
+_MESSAGE = ("This will migrate the Shotgun Desktop configuration away from the 'Template "
+            "Project'. Before running this, make sure all your users have installed version 1.2.0 or "
+            "greater of the Shotgun Desktop or that they are running version 1.2.0 or greater of the Startup framework. "
+            "You can see which versions you are running in the Shotgun Desktop's About Box.")
+
```

Comment:
Just a couple of suggestions for the text:

- First sentence: This will migrate the Shotgun site configuration used by the Desktop app so it is no longer associated with the 'Template Project'.

- For the version check, should we just mention the startup framework?  Since desktop v1.1.0 with the update would be valid as well?  We should also say that if you don't see the version of the startup framework in the about box then you are running an older version.

- I think an additional sentence to stress that this migration will break users on the old code would be good.  Something like.  "WARNING: If there are people using older versions of Desktop with your site, this migration will result in them getting an error on Desktop startup."